### mission params ###
mission:
  # waypoint units in [[m], [m], [rad]]
  waypoints:
    [
      # # # # test5 scenario
      # [0.0,   0.0,   0.0], # large circle start
      # [3.87,  0.3,   0.0],
      # [3.72,  -0.25,  1.54],
      # [3.11,  -0.95,   3.14],
      # [1.57,  -1.0,  3.11],
      # [0.0,   0.0,   3.14],
      # [0.0,   0.0,   0.0],
      # [3.87,  0.0,   0.0],
      # [3.72,  -0.3,  1.54],
      # [3.11,  -1.05,   3.14],
      # [1.57,  -0.85,  3.11],
      # [0.0,   0.0,   3.14],
      # [0.0,   0.0,   0.0], # large circle start
      # [3.87,  0.0,   0.0],
      # [3.72,  -0.25,  1.54],
      # [3.11,  -0.95,   3.14],
      # [1.57,  -1.0,  3.11],
      # [0.0,   0.0,   3.14],
      # [0.0,   0.0,   0.0],
      # [3.87,  0.0,   0.0],
      # [0.3,   0.0,   0.0],
      # [0.3,   0.0,   1.54],
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 1 circle done
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 2 circle done
      # [0.0,   0.0,   3.14], 
      # [2.3,   0.0,   3.14],
      # [0.0,   0.0,   3.14],

      # # test4 scenario
      [0.3,   0.0,   0.0],
      [0.3,   0.0,   1.54],
      [0.51,  0.49,  0.75],
      [1.0,   0.7,   0.0],
      [1.5,   0.49,  -0.75],
      [1.7,   0.0,   -1.54],
      [1.5,  -0.49, -2.3],
      [1.0,  -0.7,   3.14],
      [0.51, -0.49,  2.3],
      [0.3,   0.0,   1.54], # 1 circle done
      [0.51,  0.49,  0.75],
      [1.0,   0.7,   0.0],
      [1.5,   0.49,  -0.75],
      [1.7,   0.0,   -1.54],
      [1.5,  -0.49, -2.3],
      [1.0,  -0.7,   3.14],
      [0.51, -0.49,  2.3],
      [0.3,   0.0,   1.54], # 2 circle done
      [0.0,   0.0,   3.14], 
      [2.3,   0.0,   3.14],
      [0.0,   0.0,   3.14],
      [0.0,   0.0,   0.0], # large circle start
      [3.87,  0.0,   0.0],
      [3.72,  -0.25,  1.54],
      [3.11,  -0.95,   3.14],
      [1.57,  -1.0,  3.11],
      [0.0,   0.0,   3.14],
      [0.0,   0.0,   0.0],
      [3.87,  0.0,   0.0],

      # # # test3_fixed scenario
      # [0.3,   0.0,   0.0],
      # [0.3,   0.0,   1.54],
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 1 circle done
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 2 circle done
      # [0.0,   0.0,   3.14], 
      # [2.3,   0.0,   3.14],
      # [0.0,   0.0,   3.14],
      # [0.0,   0.0,   0.0],
      # [3.87,  0.0,   0.0],
      # [3.87,  0.0,   3.14],
      # [0.0,   0.0,   0.0],

      # # # test2 scenario
      # [0.3,   0.0,   0.0],
      # [0.3,   0.0,   1.54],
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 1 circle done
      # [0.51,  0.49,  0.75],
      # [1.0,   0.7,   0.0],
      # [1.5,   0.49,  -0.75],
      # [1.7,   0.0,   -1.54],
      # [1.5,  -0.49, -2.3],
      # [1.0,  -0.7,   3.14],
      # [0.51, -0.49,  2.3],
      # [0.3,   0.0,   1.54], # 2 circle done
      # [0.0,   0.0,   3.14], 

      # # test1 scenario - drive back and forth
      # [2.3, 0.0, 0.0],
      # [0.0, 0.0, 0],
      # [3.87, 0.0, 0.0],
      # [3.87, 0.0, 1.57],
      # [3.87, 0.0, 3.14],
      # [3.87, 0.0, -1.57],
      # [3.87, 0.0, 0.0],
      # [0.0, 0.0, 0],
      # [0.0, 0.0, 3.14],
      # [3.87, 0.0, 3.14],
      # [0.0, 0.0, 3.14]
    ]
  # target_angle: -1.57 # Robot rotates to this angle at the end. From -PI to PI [rad]
  distance_margin: 0.05   # [m]
  heading_margin: 0.01 # [rad]
### controller params ###
controller:
  # controller gains
  Kp: [0.5, 0.5, 0.5]  # proportional gains [x, y, angular]
  Kd: [0.1, 0.1, 0.1]  # derivative gains [x, y, angular]
  Ki: [0.01, 0.01, 0.01]  # derivative gains [x, y, angular]
  vel_limits_xy: 0.06
  vel_limits_theta: [-0.22, 0.22]